_drafts/contribute-to-ansible.markdown

```diff
@@ -183,39 +183,243 @@ $ source hacking/env-setup
+
+Example:
+
+Considere you want to works on the docker features of ansible, for fixing a bug or implement a new feature,
+you research for appropriated label and find the existing label docker. So apply this filter to your research
+and reach only the issues in connection with docker subject.
+
+![Ansible filter issues by labels]({{ "/assets/contribute-to-ansible/ansible-filter-by-labels.png" | absolute_url }})
+
+Also you can filter for kind of issues (fix or feature) and refine your research.
+
+#### b. Start to work
+
+Well! You have your developmenent environment, you know basicly how to work on ansible, how about actually starting to work?
+
+On the ansible project the default git branch is devel, devel correspond to the latest version of this product.
+Ansible ask to you to work on the `devel` branch by creating a new local feature branch based on this.
+Backports between versions of ansible as managed by the core team.
+
+Considere you want to fix a bug on docker module so the workflow looks like this:
+
+```sh
+$ git checkout devel
+$ git checkout -b fix-docker-module
+```
+
+You are now on your new branch `fix-docker-module`. You can start to work on and to try to resolve your bug.
+
+Patches should always be made against the `devel` branch.
+
+For more informations you can read the
+[official ansible contributing guide](http://docs.ansible.com/ansible/latest/community.html#contributing-code-features-or-bugfixes).
+
+Now you have start to work you must becareful about coding rules best practices.
+
+#### c. Apply coding rules best practices
+
+Ansible was maintained by lot of people so this project need to impose somes 
+development guidelines for to ensure project quality, stability, and maintainability.
+
+Ansible’s aesthetic encourages simple, readable code and consistent,
+conservatively extending, backwards-compatible improvements.
+When contributing code to Ansible, please observe the following guidelines:
+
+- Code developed for Ansible needs to support Python2-2.6 or higher and Python3-3.5 or higher.
+- Use a 4-space indent, not tabs.
+- Ansible team do not enforce 80 column lines; up to 160 columns are acceptable.
+- Ansible team do not accept ‘style only’ pull requests unless the code is nearly unreadable.
+- Ansible are not strictly compliant with [PEP8](https://www.python.org/dev/peps/pep-0008/).See [PEP8](https://www.python.org/dev/peps/pep-0008/) for more information.
+
+Ansible are not strictly compliant with PEP8 but we still want to encourage you to respect this standard.
+
+More informations about coding rules are availables [here](http://docs.ansible.com/ansible/latest/community.html#for-current-and-prospective-developers).
+
+#### d. Test your works
+
+Ansible was battery included and offer to you a many functionalities very useful during your developments.
+Ansible comes with a complet suite of tests what you must use to ensure what you don't introduce bug, or broken an existing functionalities.
```

Comment:
Ansible comes with a full suite of tests that avoiding to introduce bugs or breaking some functionalities. You so have to pass these tests and should write or improve existing if you find a lack on somes . You do not need an issue for that, feel free to contribute on it, it will be a great improvement.